src/components/SlateEditor/plugins/mathml/MathML.tsx

```diff
@@ -7,39 +7,63 @@
  */
 
 import React, { useState, useEffect, useRef } from 'react';
-import { RenderElementProps } from 'slate-react';
+import { Editor } from 'slate';
+import { ReactEditor } from 'slate-react';
+import { MathmlElement } from '.';
 
 interface Props {
-  attributes: RenderElementProps['attributes'];
   model: {
     xlmns: string;
     innerHTML: string;
   };
+  editor: Editor;
+  element: MathmlElement;
 }
 
-const MathML = ({ model }: Props) => {
+const MathML = ({ model, element, editor }: Props) => {
   const [reRender, setReRender] = useState(false);
+  const [mathjaxInitialized, setMathjaxInitialized] = useState(true);
 
   const mounted = useRef(false);
+
   useEffect(() => {
     if (!mounted.current) {
       mounted.current = true;
     } else {
+      const { MathJax } = window;
+      const node = ReactEditor.toDOMNode(editor, element);
+      if (MathJax && node) {
+        MathJax.typesetClear([node]);
+      }
       // Note: a small delay before a 're-render" is required in order to
       // get the MathJax script to render correctly after editing the MathML
       setReRender(true);
       setTimeout(() => {
         setReRender(false);
+        setMathjaxInitialized(false);
       }, 10);
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [model.innerHTML]);
 
+  useEffect(() => {
+    if (mathjaxInitialized) {
+      return;
+    }
+    const { MathJax } = window;
+    const node = ReactEditor.toDOMNode(editor, element);
+    if (MathJax && node) {
+      MathJax.typeset([node]);
```

Comment:
Er ikkje dette bedre i og med at det er async?
```suggestion
      MathJax.typesetPromise([node]);
```

Reply:
Tror ikke det har så mye å si, men det anbefales at man unngår at flere instanser av typeset parallelt. Så om vi bruker `typeset()` er det hvertfall mindre sannsynlig at dette skjer.